src/JustSaying/AwsTools/MessageHandling/SqsNotificationListener.cs

```diff
@@ -93,7 +92,7 @@ public SqsNotificationListener WithMessageProcessingStrategy(IMessageProcessingS
             return this;
         }
 
-        public void AddMessageHandler<T>(Func<IHandlerAsync<T>> futureHandler) where T : Message
+        public void AddMessageHandler<T>(Func<IHandlerAsync<T>> futureHandler, Func<T, string> uniqueKeySelector = default) where T : class
```

Comment:
If this is optional, how does it avoid causing the `ArgumentNullException` later? Or is this causing the test failure you mentioned?

Reply:
Wow. I didn't know there was a `default` for `Func<T, string>`!

Reply:
There isn't, well there is, it's just `null`.

Reply:
Oh, haha. I don't know why I assumed otherwise.

Reply:
`default` is just syntactic sugar added in C# 8 (at least I think it was 8, maybe a minor revision of 7) for `default(Type)`. It's `null` for any reference type and the "zero" value for value types (`0`, `0l, etc.). The same behaviour you'd get from LINQ's `FirstOrDefault()`.